lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js

```diff
@@ -7,11 +7,31 @@ module.exports = {
   compileRestApi() {
     this.apiGatewayRestApiLogicalId = this.provider.naming.getRestApiLogicalId();
 
+    let endpointType = 'EDGE';
+
+    if (this.serverless.service.provider.endpointType) {
+      const validEndpointTypes = ['REGIONAL', 'EDGE'];
```

Comment:
According to the AWS docs, an endpoint type can also be a domain name. Should we support this? I think it could be important as soon as you map a custom domain names to your API - but I did not test anything in that area yet.

Reply:
@HyperBrain I think it's slightly different than that. A [domain name that you register with API Gateway](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-domainname.html#cfn-apigateway-domainname-endpointconfiguration) can also have an endpoint configuration of Edge / Regional. Thus, if you set up `api.mycompany.com` in API Gateway, you could configure it to be Regional.

Currently, the Framework doesn't configure API GW domains in core, so I don't think we can do anything with that. It could be a feature for the [serverless-domain-manager](https://github.com/amplify-education/serverless-domain-manager) plugin to add.

I'm not 100% sure I'm correct on this so feel free to correct me :)

Reply:
The reason why I came to the assumption that it could be defined here is, that the CF AWS::ApiGateway::RestApi resource defines its `EndpointConfiguration` property here in that way: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-restapi-endpointconfiguration.html

There is written "... or its custom domain name". Additionally it seems that the property (which is an array) can contain multiple values. But don't ask me how exactly this is intended to work or when exactly you should add multiple values to the array.

So this is for me a bit unclear too 😄

Reply:
Good point. That's confusing.

For what it's worth, the [EndpointConfiguration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-restapi-endpointconfiguration.html) documentation for the RestApi resource is nearly identical to the [EndpointConfiguration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-domainname-endpointconfiguration.html) for the DomainName resource. I wonder if that sentence -- `A list of endpoint types of an API or its custom domain name` refers to the fact that this configuration is similar for both a RestApi resource and a DomainName resource?

Agreed that the array makes no sense 😄